README.md

```diff
@@ -71,18 +62,294 @@ test({
 });
 ```
 
+`SerializeProperty` is the
+[property decorator](https://www.typescriptlang.org/docs/handbook/decorators.html#property-decorators)
+used before a property to add the property to the serialization logic.
+
+```ts
+import { Serializable, SerializeProperty } from "./mod.ts";
+import { assertEquals, test } from "./test_deps.ts";
+
+test({
+  name: "SerializeProperty adds to serialization logic",
+  fn() {
+    class Test extends Serializable {
+      @SerializeProperty()
+      testName = "toJSON";
+    }
+    assertEquals(new Test().toJSON(), `{"testName":"toJSON"}`);
+    const testObj = new Test().fromJSON(`{"testName":"fromJSON"}`);
+    assertEquals(testObj.testName, "fromJSON");
+  },
+});
+```
+
+### Methods
+
+Each method has an implementable interface if you wish to provide your own
+functionality.
+
+#### `fromJSON`
+
+Takes one argument, the JSON `string` or `JSONObject` to deserialize creating an
+object. `fromJSON` will perform provided `tsTransformKey` operations and
+strategy value transformations.
+
+```ts
+import { Serializable } from "./mod.ts";
+import { assert, test } from "./test_deps.ts";
+
+test({
+  name: "Method fromJSON",
+  fn() {
+    class TestClass extends Serializable {}
+    const testObjOne = new TestClass().fromJSON({});
+    const testObjTwo = new TestClass().fromJSON("{}");
+    assert(testObjOne instanceof TestClass);
+    assert(testObjTwo instanceof TestClass);
+  },
+});
+```
```

Comment:
While I think the doc tests are very useful for our purposes they unfortunately make the readme more difficult to read. If I am a user who is looking up how to parse JSON into a class they will be presented with an entire test case rather than just the code they need to get started.

We may want to consider hiding the boilerplate and assertion code somehow, possibly some kind of markdown template which can generate a "cleaned" markdown file?

For example, given the code example:
```ts
// #md-template-start
import { Serializable } from "./mod.ts";
import { assert, test } from "./test_deps.ts";
test({
  name: "Method fromJSON",
  fn() {
    // #md-code-start
    class TestClass extends Serializable {}
    
    const testObjOne = new TestClass().fromJSON({});
    // or
    const testObjTwo = new TestClass().fromJSON("{}");
    // #md-code-end
    assert(testObjOne instanceof TestClass);
    assert(testObjTwo instanceof TestClass);
  },
});
// #md-template-end
```

would produce

```ts
class TestClass extends Serializable {}

const testObjOne = new TestClass().fromJSON({});
// or
const testObjTwo = new TestClass().fromJSON("{}");
```

Not sure if we want to fix it here, or possibly as it's own issue.

Reply:
great point, the code examples are not examples...

What if we remove the testing bits and we can `deno run` instead of test in the script

Reply:
That sounds good to me, I think having regular tests to do the testing should be sufficient coverage, and being able to run the code examples without errors should be enough